### PR TITLE
fix: allows other algorithms to be used during the verification process

### PR DESCRIPTION
**Pull request by mineme0110**
The current implementation defaults to the ES256 algorithm, causing other keys with different signature algorithms to fail during the verification process of the SDJWT.
This fix will make the signature algorithm flexible.
This flexibility will allow other algorithms, such as EdDSA, RSA etc to work.

### CODE REVIEW

**Comment by mineme0110** [comment]
@jovfer  Please can you review this

**Comment by mineme0110** [comment]
@tkuhrt Please can you review this request

**Review by elribonazo — APPROVED** [review]
LGTM, this provides the implementation with a little more flexibility on the crypto primitives that want to be used for sd+jwt. 

Thanks for adding that @mineme0110

**Review by hamada147 — APPROVED** [review]

**Review by goncalo-frade-iohk — APPROVED** [review]

**Review by jovfer — APPROVED** [review]

File: src/lib.rs [hunk 1]
```diff
@@ -179,12 +181,14 @@ impl SDJWTCommon {
...
-        self.unverified_sd_jwt = Some(format!(
+        let sd_jwt = format!(
             "{}.{}.{}",
             parsed_sd_jwt_json.protected,
             parsed_sd_jwt_json.payload,
             parsed_sd_jwt_json.signature
-        ));
+        );    
+        self.unverified_sd_jwt = Some(sd_jwt.clone());
+        self.sign_alg = Self::decode_header_and_get_sign_algorithm(&sd_jwt);
```
[the remark this answers is not in the document]
Comment (jovfer) [reply src/lib.rs:191]: TBD in separate PR

**Pull request merged**